### DISCUSSION TITLE
How do I get instance objects on a component

### DISCUSSION THREAD

**Discussion by ResponseState200** [vuejs/router#1870] [Help and Questions] [answered]
### Reproduction

none

### Steps to reproduce the bug

router.afterEach((to, from) => {
  to.matched[0].instances.default;
  onst routeComponent = router.resolve(to.path).route.component
})

### Expected behavior

Gets the instance of the rendered component under the route

### Actual behavior

I tried a lot of things but I couldn't get it

**Comment by posva** [comment, marked as answer]
Instances are a private API, they are used internally for in-component navigation guards. In practice is better off not using them in application code as it leads to non typed and fragile coupling
